receiver/scraper_test.go

```diff
@@ -0,0 +1,183 @@
+		ProcessRuntimeMemstatsFrees:         metricEnabled,
+		ProcessRuntimeMemstatsGcCPUFraction: metricEnabled,
+		ProcessRuntimeMemstatsGcSys:         metricEnabled,
+		ProcessRuntimeMemstatsHeapAlloc:     metricEnabled,
+		ProcessRuntimeMemstatsHeapIdle:      metricEnabled,
+		ProcessRuntimeMemstatsHeapInuse:     metricEnabled,
+		ProcessRuntimeMemstatsHeapObjects:   metricEnabled,
+		ProcessRuntimeMemstatsHeapReleased:  metricEnabled,
+		ProcessRuntimeMemstatsHeapSys:       metricEnabled,
+		ProcessRuntimeMemstatsLastPause:     metricEnabled,
+		ProcessRuntimeMemstatsLookups:       metricEnabled,
+		ProcessRuntimeMemstatsMallocs:       metricEnabled,
+		ProcessRuntimeMemstatsMcacheInuse:   metricEnabled,
+		ProcessRuntimeMemstatsMcacheSys:     metricEnabled,
+		ProcessRuntimeMemstatsMspanInuse:    metricEnabled,
+		ProcessRuntimeMemstatsMspanSys:      metricEnabled,
+		ProcessRuntimeMemstatsNextGc:        metricEnabled,
+		ProcessRuntimeMemstatsNumForcedGc:   metricEnabled,
+		ProcessRuntimeMemstatsNumGc:         metricEnabled,
+		ProcessRuntimeMemstatsOtherSys:      metricEnabled,
+		ProcessRuntimeMemstatsPauseTotal:    metricEnabled,
+		ProcessRuntimeMemstatsStackInuse:    metricEnabled,
+		ProcessRuntimeMemstatsStackSys:      metricEnabled,
+		ProcessRuntimeMemstatsSys:           metricEnabled,
+		ProcessRuntimeMemstatsTotalAlloc:    metricEnabled,
+	}
+	allMetricsDisabled = metadata.MetricsSettings{
+		ProcessRuntimeMemstatsBuckHashSys:   metricDisabled,
+		ProcessRuntimeMemstatsFrees:         metricDisabled,
+		ProcessRuntimeMemstatsGcCPUFraction: metricDisabled,
+		ProcessRuntimeMemstatsGcSys:         metricDisabled,
+		ProcessRuntimeMemstatsHeapAlloc:     metricDisabled,
+		ProcessRuntimeMemstatsHeapIdle:      metricDisabled,
+		ProcessRuntimeMemstatsHeapInuse:     metricDisabled,
+		ProcessRuntimeMemstatsHeapObjects:   metricDisabled,
+		ProcessRuntimeMemstatsHeapReleased:  metricDisabled,
+		ProcessRuntimeMemstatsHeapSys:       metricDisabled,
+		ProcessRuntimeMemstatsLastPause:     metricDisabled,
+		ProcessRuntimeMemstatsLookups:       metricDisabled,
+		ProcessRuntimeMemstatsMallocs:       metricDisabled,
+		ProcessRuntimeMemstatsMcacheInuse:   metricDisabled,
+		ProcessRuntimeMemstatsMcacheSys:     metricDisabled,
+		ProcessRuntimeMemstatsMspanInuse:    metricDisabled,
+		ProcessRuntimeMemstatsMspanSys:      metricDisabled,
+		ProcessRuntimeMemstatsNextGc:        metricDisabled,
+		ProcessRuntimeMemstatsNumForcedGc:   metricDisabled,
+		ProcessRuntimeMemstatsNumGc:         metricDisabled,
+		ProcessRuntimeMemstatsOtherSys:      metricDisabled,
+		ProcessRuntimeMemstatsPauseTotal:    metricDisabled,
+		ProcessRuntimeMemstatsStackInuse:    metricDisabled,
+		ProcessRuntimeMemstatsStackSys:      metricDisabled,
+		ProcessRuntimeMemstatsSys:           metricDisabled,
+		ProcessRuntimeMemstatsTotalAlloc:    metricDisabled,
+	}
+)
+
+func newMockServer(t *testing.T, responseBodyFile string) *httptest.Server {
```

Comment:
```suggestion
func newMockServer(tb testing.TB, responseBodyFile string) *httptest.Server
```
Allows you to use it in either benchmarks or testing.

Reply:
Done (I committed it myself so I didn't have to pull the change down)